internal/sqlquery/scraper.go

```diff
@@ -107,3 +109,142 @@ func (s *Scraper) Shutdown(_ context.Context) error {
 	}
 	return nil
 }
+
+func BuildDataSourceString(driver string, dataSource DataSourceConfig) (string, error) {
```

Comment:
I'm seeing a lot of duplicated logic across the `build...String(dataSource)` methods. Can we try to consolidate? For example, building the auth string and query parameters looks just about identical, as well as appending the query (if there is one).

Maybe we have all of the logic in this method, then the switch is simply formatting the connection string for each driver, rather than an individual method.